Plugins/Wox.Plugin.Folder/Main.cs

```diff
@@ -37,89 +36,100 @@ public Control CreateSettingPanel()
-                        },
-                        ContextData = item,
-                    }).ToList();
+            var results = GetUserFolderResults(query);
 
+            string search = query.Search.ToLower();
             if (_driverNames != null && !_driverNames.Any(search.StartsWith))
                 return results;
 
-            //if (!input.EndsWith("\\"))
-            //{
-            //    //"c:" means "the current directory on the C drive" whereas @"c:\" means "root of the C drive"
-            //    input = input + "\\";
-            //}
             results.AddRange(QueryInternal_Directory_Exists(query));
 
-            // todo temp hack for scores
+            // todo why was this hack here?
             foreach (var result in results)
             {
                 result.Score += 10;
             }
 
             return results;
         }
+
+        private Result CreateFolderResult(string title, string path)
+        {
+            return new Result
+            {
+                Title = title,
+                IcoPath = path,
+                SubTitle = "Ctrl + Enter to open the directory",
+                Action = c =>
+                {
+                    if (c.SpecialKeyState.CtrlPressed)
+                    {
+                        try
+                        {
+                            Process.Start(path);
+                            return true;
+                        }
+                        catch (Exception ex)
+                        {
+                            MessageBox.Show(ex.Message, "Could not start " + path);
+                            return false;
+                        }
+                    }
+
+                    string changeTo = path.EndsWith("\\") ? path : path + "\\";
+                    _context.API.ChangeQuery(changeTo);
+                    return false;
+                }
+            };
+        }
+
+        private List<Result> GetUserFolderResults(Query query)
+        {
+            string search = query.Search.ToLower();
+            var userFolderLinks = _settings.FolderLinks.Where(
+                x => x.Nickname.StartsWith(search, StringComparison.OrdinalIgnoreCase));
+            var results = userFolderLinks.Select(item => CreateFolderResult(item.Nickname, item.Path))
+                    .ToList();
+            return results;
+        }
+
         private void InitialDriverList()
         {
             if (_driverNames == null)
             {
                 _driverNames = new List<string>();
-                DriveInfo[] allDrives = DriveInfo.GetDrives();
+                var allDrives = DriveInfo.GetDrives();
                 foreach (DriveInfo driver in allDrives)
                 {
                     _driverNames.Add(driver.Name.ToLower().TrimEnd('\\'));
                 }
             }
         }
 
+        private static readonly char[] _specialSearchChars = new char[]
+        {
+            '?', '*', '>'
```

Comment:
@theClueless how is the character '?' used? I have yet to be able to activate it.

Reply:
it is a wildcard for one character instead of many, example:
![image](https://user-images.githubusercontent.com/14300910/70515672-64b92900-1b3e-11ea-8877-985a23f5b687.png)

so here I'm searching for something that has 2 chars and then mp and it will find all that applay